guides/common/modules/proc_logging-in-to-project-configured-with-keycloak-as-an-authentication-source.adoc

```diff
@@ -0,0 +1,47 @@
+[id="logging-in-to-project-configured-with-keycloak-as-an-authentication-source_{context}"]
+= Logging in to {Project} configured with {keycloak} as an authentication source
+
+With {keycloak} configured as an external authentication source for {Project}, users defined in a {keycloak} realm can log in to {ProjectServer}.
+The particular login methods available to users depend on how you configured integration between {keycloak} and {Project}.
+
+.Procedure
+
+To authenticate to the {ProjectWebUI}:
+
+* In your browser, log in to {Project} and enter your credentials.
+
+To authenticate to the Hammer CLI:
+
+. To authenticate to the {Project} CLI using the code grant type, enter the following command:
++
+[options="nowrap", subs="+quotes,attributes"]
+----
+# hammer auth login oauth \
+--two-factor \
+--oidc-token-endpoint 'https://_{keycloak-example-com}_/auth/realms/ssl-realm/protocol/openid-connect/token' \
+--oidc-authorization-endpoint 'https://_{keycloak-example-com}_/auth' \
+--oidc-client-id '_{foreman-example-com}_-foreman-openidc' \
+--oidc-redirect-uri urn:ietf:wg:oauth:2.0:oob
+----
++
+You will be prompted to enter a success code.
+. To retrieve the success code, navigate to the URL that the command returns.
+. Enter the success code in CLI.
```

Comment:
Yuck, this is ugly. Can Hammer be configured to do these steps programmatically, I wonder? Non-blocking to this PR.

Reply:
The reason this is done this way is that you may run hammer on a machine where there is no graphical client. Opening a browser via SSH can be done in some circumstances, but it's certainly not the default scenario.